File: arrays/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        charCount = {}

        for letter in s:
            charCount[letter] = charCount.get(letter, 0) + 1

        for letter in t:
            if letter not in charCount:
                return False
            charCount[letter] -= 1

        for v in charCount.values():
            if v != 0:
                return False

        return True

class HackySolution:
    def isAnagram(self, s: str, t: str) -> bool:
        return Counter(s) == Counter(t)

File: arrays/valid_anagram.cpp
class Solution {
public:
  bool isAnagram(string s, string t) {

    if (s.length() != t.length())
      return false;

    std::unordered_map<char, int> dic;
    int n = s.length();

    for (int i = 0; i < n; i++) {
      dic[s[i]]++;
      dic[t[i]]--;
    }

    for (const auto &[key, value] : dic) {
      if (value != 0) {
        return false;
      }
    }
    return true;
  }
};
